src/components/GlobalFilter/GlobalFilter.tsx

```diff
@@ -1,123 +1,176 @@
 import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react';
-import { batch, shallowEqual, useDispatch, useSelector } from 'react-redux';
-import { useTagsFilter } from '@redhat-cloud-services/frontend-components/FilterHooks';
+import { AllTag, useTagsFilter } from '@redhat-cloud-services/frontend-components/FilterHooks';
 import debounce from 'lodash/debounce';
-import { fetchAllSIDs, fetchAllTags, fetchAllWorkloads, globalFilterChange } from '../../redux/actions';
 import { generateFilter } from './globalFilterApi';
 import { useLocation, useNavigate } from 'react-router-dom';
-import { GlobalFilterDropdown, GlobalFilterDropdownProps } from './GlobalFilterMenu';
+import { GlobalFilterDropdown } from './GlobalFilterMenu';
 import { storeFilter } from './filterApi';
-import { GlobalFilterTag, GlobalFilterWorkloads, ReduxState, SID } from '../../redux/store';
-import { FlagTagsFilter } from '../../@types/types';
+
 import { isGlobalFilterAllowed } from '../../utils/common';
 import InternalChromeContext from '../../utils/internalChromeContext';
 import ChromeAuthContext from '../../auth/ChromeAuthContext';
-import { useAtomValue } from 'jotai';
-import { activeModuleAtom } from '../../state/atoms/activeModuleAtom';
+import { useAtomValue, useSetAtom } from 'jotai';
+import {
+  globalFilterHiddenAtom,
+  isDisabledAtom,
+  isLoadedAtom,
+  registeredWithAtom,
+  selectedTagsAtom,
+  sidsAtom,
+  tagsAtom,
+  workloadsAtom,
+} from '../../state/atoms/globalFilterAtom';
+import { getAllSIDs, getAllTags, getAllWorkloads } from './tagsApi';
 
-const useLoadTags = (hasAccess = false) => {
+const useLoadTags = (
+  hasAccess = false,
+  setTags: ReturnType<typeof useSetAtom<typeof tagsAtom>>,
+  setSids: ReturnType<typeof useSetAtom<typeof sidsAtom>>,
+  setWorkloads: ReturnType<typeof useSetAtom<typeof workloadsAtom>>
+) => {
   const navigate = useNavigate();
-  const registeredWith = useSelector(({ globalFilter: { scope } }: ReduxState) => scope);
-  const activeModule = useAtomValue(activeModuleAtom);
-  const isDisabled = useSelector(({ globalFilter: { globalFilterHidden } }: ReduxState) => globalFilterHidden || !activeModule);
-  const dispatch = useDispatch();
+  const registeredWith = useAtomValue(registeredWithAtom);
+  const isDisabled = useAtomValue(isDisabledAtom);
   return useCallback(
-    debounce((activeTags: any, search: any) => {
-      storeFilter(activeTags, hasAccess && !isDisabled, navigate);
-      batch(() => {
-        dispatch(
-          fetchAllTags({
+    debounce(async (activeTags: any, search: any) => {
+      // Set loading state to false before fetching
+      setTags((prev) => ({ ...prev, isLoaded: false }));
+      setSids((prev) => ({ ...prev, isLoaded: false }));
+      setWorkloads((prev) => ({ ...prev, isLoaded: false }));
```

Comment:
When we do multiple "set calls", each of them means one render. I'd suggest creating a write atom that would set the loading with the same value (in this case false) state to all of these individual atoms.

It can be re-used down bellow.